### PR TITLE
Renamed FileStreamResult to StreamResult

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #2787 
@Eilon

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/FileStreamResult.cs [hunk 1]
```diff
@@ -13,48 +13,48 @@ namespace Microsoft.AspNet.Mvc
 {
     /// <summary>
     /// Represents an <see cref="ActionResult"/> that when executed will
-    /// write a file from a stream to the response.
+    /// write a stream to the response.
     /// </summary>
-    public class FileStreamResult : FileResult
+    public class StreamResult : FileResult
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/FileStreamResult.cs]: I think this name is confusing...for users of Web API they might translate `StreamContent` -> `StreamResult` just like how `ObjectContent` became `ObjectResult`

**Comment by davidfowl** [comment]
Why is the build failing?

File: src/Microsoft.AspNet.Mvc.Core/FileStreamResult.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNet.Mvc.Core/FileStreamResult.cs]: Why does it derive from FileResult?

Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/FileStreamResult.cs]: Instead should we have `StreamResult` and also `FileStreamResult` which would derive from `StreamResult`?
@rynowak

**Comment by ajaybhargavb** [comment]
I don't think the build failures are related to this change.

**Comment by Eilon** [comment]
@davidfowl The build failures are do to aspnetvnext being behind volatile so the new packages from Hao aren't there yet.

**Comment by Eilon** [comment]
Today we have:


Base class:
```c#
abstract FileResult
	protected abstract Task WriteFileAsync(HttpResponse response, CancellationToken cancellation);
```

Three concrete classes:
```c#
FileContentResult : FileResult
FilePathResult : FileResult
FileStreamResult : FileResult
```

The idea of the "file" base results is that they are meant for rendering a "file" and I think optionally prompting a "Save As" dialog in the browser.

Is there a better set of names we can use?

**Comment by davidfowl** [comment]
So these things automatically set the content disposition as well right? We should have a pure `StreamResult` and have a different `FileStreamResult` that actually takes a file stream since they are different things.

**Comment by Eilon** [comment]
If you set the `FileDownloadName` property on the result it'll set the `Content-Disposition` header. Otherwise it just dumps the file as-is (e.g. maybe you're rendering an image from a database).

We could certainly just add a new `StreamResult` that derives from nothing special.

Would there be any reason to specifically support `FileStream` though? The `FileStreamResult` action result takes in `Stream`, which is what `FileStream` derives from. What would we do differently for `FileStream`? Would we use it to grab the filename or something?

**Comment by rynowak** [comment]
Streams 'just work' even without an action result.

**Comment by Eilon** [comment]
Yes, but action result == happy unit testing :smile:

**Comment by rynowak** [comment]
> Yes, but action result == happy unit testing :smile:

`ObjectResult` then. 

I'm not opposed to adding `StreamResult` to make it more obvious. You could say that `Stream` + `ObjectResult` isn't as discoverable, but adding it doesn't address any scenarios that aren't *easy* already.

**Comment by danroth27** [comment]
Moved this to beta7. Please commit to dev, not release.

**Comment by darrelmiller** [comment]
Anyone know what happened to the plan to add StreamResult?  I just spent the last half an hour trying to figure out if FileStreamResult did anything funky as I just want to return a stream and a content type.

@rynowak Because of the crazy stuff ObjectContent does, I would never have thought of using ObjectResult.

**Comment by rynowak** [comment]
FileStreamResult should do what you want. It can do ranges by default, and it can optionally do ETag/If-ModifiedSince and Content-Disposition

**Comment by darrelmiller** [comment]
@rynowak Thanks.  Feels a bit weird but I'll get used to it.

**Comment by davidfowl** [comment]
It's pretty weird. We *could* add `StreamResult` that did the same thing as `FileStreamResult` to improve discoverability.

**Pull request closed without merging**